reports/models.py

```diff
@@ -0,0 +1,459 @@
+        reports = self.unsynced(sync_from=sync_from)
+        for report in reports:
+            report._rehydrate(hydrate_linkedevents_event=hydrate_linkedevents_event)
+            report.save()
+        return reports
+
+
+class EnrolmentReport(TimestampedModel):
+    """
+    Enrolment report model is used to offer some exportable data related to enrolments.
+    It stores the information about enrolments and their study groups,
+    the coordinates of the group unit position and the coordinates of
+    the event position to calculate the distance to event,
+    the event data like time frame of the event and te publisher,
+    capacity information, event keywords, etc.
+    """
+
+    # Study group
+    # - Store the foreign key to fetch all data
+    # - If the foreign key integrity is gone,
+    # keep some information so the data is still available for reports:
+    #     1. Group sizes: Kids and adults
+    #     2. Study levels
+    # - Store the unit position (coordinates) and divisions to normalise the data.
+    # Place data must be fetched from LinkedEvents API
+    # (or from other service like the servicemap API)
+
+    _study_group = models.ForeignKey(
+        occurrences_models.StudyGroup,
+        on_delete=models.SET_NULL,
+        null=True,
+        db_column="study_group_id",
+    )
+    study_group_unit_id = models.CharField(
+        max_length=255, verbose_name=_("study group's unit id"), null=True
+    )
+    study_group_amount_of_children = models.PositiveSmallIntegerField(
+        verbose_name=_("amount of the children in the study group")
+    )
+    study_group_amount_of_adult = models.PositiveSmallIntegerField(
+        verbose_name=_("amount of the adults in the study group"), default=0
+    )
+    study_group_study_levels = ArrayField(
+        # ArrayField Size 2: Study level id, translated name
+        ArrayField(models.CharField(max_length=255), size=2),
+        verbose_name=_("study levels of the study group"),
+    )
+    # The position of the (unit) place from LinkedEvents API
+    study_group_unit_position = ArrayField(
+        models.DecimalField(max_digits=9, decimal_places=6),
+        size=2,
+        null=True,
+        verbose_name=_("coordinates of the study group unit"),
+    )
+    # The divisions of the (unit) place from LinkedEvents API
+    study_group_unit_divisions = ArrayField(
+        models.CharField(max_length=200),
+        verbose_name=_("study group unit divisions"),
+        help_text=_("Open Civic Data identifiers"),
+        null=True,
+    )
+
+    # Enrolment
+    # - Store the foreign key to fetch all data
+    # - If the foreign key integrity is gone,
+    # keep some information so the data is still available for reports:
+    #     1. enrolment time
+    #     2. enrolment status
+    # - Enrolment status is also used to check whether or not
+    # the db row should be updated.
+    _enrolment = models.ForeignKey(
```

Comment:
If there should every only be one report line per enrolment, we could make this `OneToOneField`, which would add the appropriate DB restrictions.